### DISCUSSION TITLE
onPaginationChanged event of Pagination service not found in latest version(4.1.4)

### DISCUSSION THREAD

**Discussion by meta-shubham-joshi** [ghiscoding/Angular-Slickgrid#973] [Q&A] [answered]
In version 2.17.11 we used to have onPaginationChanged event in pagination service but in version 4.1.4 not able to see this event. So, what is the alternate or substitute for this? Also, i have found some similar subscribe events like processOnPageChanged(pageNo, event) event but not sure about this. Please help me out.

**Comment by ghiscoding** [comment, marked as answer]
all events are still in place, the only difference is that they are Custom Events now and the arguments might be in a different place in the event. I just tried and it works

```html
(onPaginationChanged)="paginationChanged($event.detail)"
```

```ts
paginationChanged(args: ServicePagination) {
    console.log('paginationChanged', args);
}
```

is giving this console log
```ts
paginationChanged {pageCount: 105, pageNumber: 2, pageSize: 5, pageSizes: Array(8), totalItems: 525, …}
```
You should take a look at the [Grid & DataView Events](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Grid-&-DataView-Events#example-with-custom-event) Wiki

**Comment by meta-shubham-joshi** [comment]
Yes it works. Thanks for the immediate response.

**Reply by ghiscoding** [reply]
I forgot to also mention the alternative would be to use [Grid State](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Grid-State-&-Preset) which also has that info

**Comment by meta-shubham-joshi** [comment]
Yes I have taken clone of your project and seen the info.
